### PR TITLE
Feature/instrumental test

### PR DESCRIPTION
**Pull request by MicahSphelele**
I have created a few unit tests to unit tests to ensure that the code runs correctly and to protect against future bugs.

### CODE REVIEW

**Review by gastsail — APPROVED** [review]
Awesome contribution

**Comment by MicahSphelele** [comment]
> Awesome contribution

@gastsail Thanks very much can you grant me write access so that I can merge my contribution.

**Comment by gastsail** [comment]
> > Awesome contribution
> 
> @gastsail Thanks very much can you grant me write access so that I can merge my contribution.

Awesome ! will merge when the lint passes 👍

**Pull request merged**
